src/Compilers/Core/Portable/EncodedStringText.cs

```diff
@@ -88,7 +92,7 @@ internal static SourceText Create(Stream stream, Encoding defaultEncoding = null
 
             try
             {
-                return Decode(stream, defaultEncoding ?? GetDefaultAnsiEncoding(), checksumAlgorithm, throwIfBinaryDetected: detectEncoding);
+                return Decode(stream, defaultEncoding ?? s_fallbackEncoding, checksumAlgorithm, throwIfBinaryDetected: detectEncoding);
```

Comment:
We don't have any unit-tests that test different default ANSI code pages. Consider manually retesting scenarios described in issues addressed by recent changes in this file, it would be unfortunate to regress them. We might still have test box with Japanese OS available.